acceptance/README.md

```diff
@@ -1,257 +1,467 @@
+```
 
-Change to the `acceptance` directory in the root of the puppet repo:
-```sh
-cd /path/to/repo/puppet/acceptance
+### Testing with Git
 ```
-Install the necessary gems with bundler:
-```sh
-bundle install
+bundle exec rake ci:test:git SHA={sha|tag|branch}
 ```
 
-Now you can get a list of test-related tasks you can run via rake:
-```sh
-bundle exec rake -T
+#### From a repo on a git server
+Though we primarily run acceptance tests against a built package, it is possible
+to run these tests with a git checkout. This is most useful when testing locally
+to speed up the feedback cycle.
+
+When testing from a github repo we need to unpack the appropriate
+[runtime archive](https://github.com/puppetlabs/puppet-runtime)
+for the platform we are testing on. These pre-built archives are stored on an
+internal server, and are currently only available to Puppet, Inc. employees.
+With these archives, we get all of the runtime dependencies that are usually
+provided as a part of the puppet agent package. This allows us to replicate
+the runtime environment produced via a package install for the purpose of
+running acceptance tests.
+
+When testing with git, `SHA` can be set to any git artifact: a long sha, a short
+sha, a tag, a branch name, etc. What happens is that we write a gemfile with the
+details of the puppet repo, pointing to the artifact referenced with `SHA`. Then
+when we run `bundle install` on the testing host, bundler grabs puppet from
+wherever the gemfile points. If the git artifact referenced is not from the
+puppetlabs repo, you can use `FORK` to point to a different github namespace.
+Likewise, if the artifact you want to access is not available on `github.com`
+but a custom git server, you can set `SERVER` to customize the git uri bundler
+pulls from. For more details on these environment variables, run
+`bundle exec rake ci:help`.
+
+As an example, if I have a development branch
+(`developent/master/major-feature`) that I'm working on and it only exists in my
+fork of puppet (`github.com/joeschmoe/puppet`), then I will run
 ```
-and view detailed information on the tasks with
-```sh
-bundle exec rake -D
+bundle exec rake ci:test:git SHA=developent/master/major-feature FORK=joeschmoe
 ```
 
-As an example, let's try running the acceptance tests using git as the code deployment mechanism.
-First, we'll have to create a beaker configuration file for a local vagrant box on which to run the tests.
-Here's what such a file could look like:
-```yaml
+Please note that any changes you want to test must be pushed up to your github
+server. This is how we access the code to be tested.
+
+#### From a local repo
+If yor are testing with git and using the docker hypervisor, you can run tests
+against the puppet checkout on your local system. You need to update your hosts
+file to add `mount_folders` to the docker host where you want the checkout of
+puppet to be available. Here, `host_path` is the path to puppet on your local
```

Comment:
i think "is the path to puppet on your local machine" is enough, and we don't need to reiterate by also having "You must make sure this matches where puppet is on your machine."